utoipa-gen/src/schema.rs

```diff
@@ -214,47 +214,42 @@ pub mod serde {
 
     use std::str::FromStr;
 
-    use proc_macro2::{Span, TokenTree};
+    use proc_macro2::{Ident, Span, TokenTree};
     use proc_macro_error::ResultExt;
     use syn::{buffer::Cursor, Attribute, Error};
 
-    #[cfg_attr(feature = "debug", derive(Debug))]
-    pub enum Serde {
-        Container(SerdeContainer),
-        Value(SerdeValue),
-    }
-
-    impl Serde {
-        #[inline]
-        fn parse_next_lit_str(next: Cursor) -> Option<(String, Span)> {
-            match next.token_tree() {
-                Some((tt, next)) => match tt {
-                    TokenTree::Punct(punct) if punct.as_char() == '=' => {
-                        Serde::parse_next_lit_str(next)
-                    }
-                    TokenTree::Literal(literal) => {
-                        Some((literal.to_string().replace('\"', ""), literal.span()))
-                    }
-                    _ => None,
-                },
+    fn parse_next_lit_str(next: Cursor) -> Option<(String, Span)> {
```

Comment:
Sorry yes I can re-add it if you like. I do think that putting inline everywhere might be overkill though? The compiler can also choose to automatically inline functions depending on optimization level (when the library user enables LTO for instance), I also thought that sometimes sometimes inlining results in worse performance, and increased compile times?

Reply:
Yeah, I think inline should work best for quite small functions. And indeed as if I have understood correctly the compiler will decide whether it will inline the function or not and will not too that necessarily, It just works as a hint to the compiler unless it is defined as `#[inline(always)]`.